tools/data_renewal_generate.py

```diff
@@ -0,0 +1,149 @@
+            if (write_header):
+                header = result.keys()
+                writer.writerow(header)
+                write_header = False
+                renewal.write("# Request for Data Collection Renewal\n")
+                renewal.write("### Renew for 1 year\n")
+                renewal.write("Total: TBD\n")
+                renewal.write("———\n")
+
+            writer.writerow(result.values())
+
+            renewal.write("`" + last_key.lstrip('.') + "`:\n")
+            renewal.write("1) Provide a link to the initial Data Collection Review Request for this collection.\n")
+            renewal.write("    - " + content["data_reviews"][0] + "\n")
+            renewal.write("\n")
+            renewal.write("2) When will this collection now expire?\n")
+            renewal.write("    - TBD\n")
+            renewal.write("\n")
+            renewal.write("3) Why was the initial period of collection insufficient?\n")
+            renewal.write("    - TBD\n")
+            renewal.write("\n")
+            renewal.write("———\n")
+            return
+
+        if type(value) is dict:
+            response(last_key + "." +  key, value, expire_version, writer, renewal)
+
+metrics_filename = "../app/metrics.yaml"
+new_metrics_filename = "../app/metrics_new.yaml"
+with open(metrics_filename, 'r') as f:
+    try:
+        arg1 = sys.argv[1]
+    except:
+        print ("usage is to include argument of the form `100`")
+        quit()
+
+    # parse metrics.yaml to json
+    write_header = True
+    data = yaml.load(f, Loader=FullLoader)
+    json_data = json.dumps(data)
+    content = json.loads(str(json_data))
+    csv_filename = arg1 + "_expiry_list.csv"
+    renewal_filename = arg1 + "_renewal_request.txt"
+    current_version = int(arg1)
+
+    # remove files created by last run if exists
+    if os.path.exists(csv_filename):
+        print("remove old csv file")
+        os.remove(csv_filename)
+
+    # remove files created by last run if exists
+    if os.path.exists(renewal_filename):
+        print("remove old renewal request template file")
+        os.remove(renewal_filename)
+
+    # remove files created by last run if exists
+    if os.path.exists(new_metrics_filename):
+        print("remove old metrics yaml file")
+        os.remove(new_metrics_filename)
+
+    data_file = open(csv_filename, 'w')
+    csv_writer = csv.writer(data_file)
+    renewal_file = open(renewal_filename, 'w')
+
+    response("", content, current_version, csv_writer, renewal_file)
+    renewal_file.close
+    print("Completed")
+    print("Total count: " + str(total_count))
+
+    # Go through the metrics.yaml file to mark expired telemetry
+    verify_count = 0
+    f.seek(0, 0)
+    data = f.readlines()
+    with open(new_metrics_filename, 'w') as f2:
+        for line in data:
+            if (line.lstrip(' ').startswith("expires: ") and not(line.lstrip(' ').startswith("expires: never"))):
+               version = int(line.lstrip(' ')[9:])
```

Comment:
Maybe leave a comment where you got this magic number from or calculate it:
```suggestion
               start_pos = len("expires: ")
               version = int(line.lstrip(' ')[start_pos:])
```

Reply:
Do you also need to try-catch the `ValueError` if the cast can fail? Are we leaving any existing telemetry to be time-based?

Reply:
Sure, I'll add the comment explaining why.  No, we don't need to try catch, metrics generation will fail if any telemetry is using date expiry.  We can either use date expiry or version expiry, Glean does not allow mixed.

Reply:
> We can either use date expiry or version expiry, Glean does not allow mixed.

TIL